packages/credential-providers/src/resolveAwsCliV2Region.ts

```diff
@@ -0,0 +1,51 @@
+import { MetadataService } from "@aws-sdk/ec2-metadata-service";
+import { logger } from "@aws-sdk/smithy-client/dist-types";
+import { loadSharedConfigFiles } from "@smithy/shared-ini-file-loader";
+
+/**
+ * Resolves the AWS region following AWS CLI V2 precedence order.
+ */
+export const resolveAwsCliV2Region = async (defaultRegion?: string, maybeProfile?: string): Promise<string> => {
```

Comment:
make parameters part of an object, like this

```js
async ({ option1, option2 }: OptionsType): Promise<string> => { ...
```

Reply:
rename `maybeProfile` to `profile`.

Reply:
the signature of this needs to be

```ts
(options) => async ({ profile? }) => string;
```

i.e., the function needs to return a function that resolves a string